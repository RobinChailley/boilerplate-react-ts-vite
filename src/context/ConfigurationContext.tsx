import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
