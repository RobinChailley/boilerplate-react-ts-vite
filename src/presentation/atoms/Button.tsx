import { useConfig } from '@context/ConfigurationContext';
import React from 'react';

interface ButtonProps {
    text: string;
    onClick: () => void;
}

const classes = {
    logout: 'w-full bg-blue-700 rounded-md p-2 text-blue-200',
};

const Button: React.FC<ButtonProps> = ({
    text,
    onClick,
}) => {
    const { translatorAdapter } = useConfig();
    const t = translatorAdapter.translate();

    return (
        <button className={classes.logout} onClick={onClick}>
            {t(text)}
        </button>
    );
};

export default Button;
