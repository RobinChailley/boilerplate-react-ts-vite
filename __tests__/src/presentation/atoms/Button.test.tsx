import { ConfigContext } from '@context/ConfigurationContext';
import Button from '@presentation/atoms/Button';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../../mocks/context/getConfigurationContextValue';

describe('Button', () => {
    it('should render correctly', async () => {
        let render = renderer.create(
            <ConfigContext.Provider value={getConfigContextValue()}>
                <Button text={''} onClick={jest.fn()}/>
            </ConfigContext.Provider>,
        );

        const snapshot = render!.toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
