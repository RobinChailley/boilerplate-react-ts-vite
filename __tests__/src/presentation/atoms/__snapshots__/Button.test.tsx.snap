// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="w-full bg-blue-700 rounded-md p-2 text-blue-200"
  onClick={[MockFunction]}
/>
`;
